Unstable sorting with mixed category imports

When basic imports include modules from multiple categories, sorting can be unstable because the sort_key isn't updated when changing order of the import names:

Test case:

```py
    def test_sorting_mixed_category_imports(self) -> None:
        self.assertUsortResult(
            """
                import os, attr
                import difflib
                import third_party
            """,
            """
                import difflib

                import attr, os
                import third_party
            """,
        )

```

Produces the following test failure:

```
(.venv) jreese@butterfree ~/workspace/usort mixed± » make format test lint
python -m ufmt format usort setup.py
python -m coverage run -m usort.tests
.......................................................................F.............
======================================================================
FAIL: test_sorting_mixed_category_imports (usort.tests.functional.UsortStringFunctionalTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jreese/workspace/usort/usort/tests/functional.py", line 632, in test_sorting_mixed_category_imports
    self.assertUsortResult(
  File "/Users/jreese/workspace/usort/usort/tests/functional.py", line 58, in assertUsortResult
    self.fail(
AssertionError: µsort result was not stable on second pass:

Before:
-------

import os, attr
import difflib
import third_party

First Pass:
-----------

import attr, os
import difflib

import third_party

Second Pass:
------------

import difflib

import attr, os

import third_party


----------------------------------------------------------------------
Ran 85 tests in 4.119s

FAILED (failures=1)
make: *** [test] Error 1
```

--- a/usort/sorting.py
+++ b/usort/sorting.py
@@ -261,6 +261,7 @@ class ImportSorter:
         # Sort items within each remaining statement
         for imp in imports:
             imp.items.sort()
+            imp.calculate_sort_key()
 
         return imports
 

--- a/usort/tests/functional.py
+++ b/usort/tests/functional.py
@@ -628,6 +628,21 @@ numpy = ["numpy", "pandas"]
             """,
         )
 
+    def test_sorting_mixed_category_imports(self) -> None:
+        self.assertUsortResult(
+            """
+                import os, attr
+                import difflib
+                import third_party
+            """,
+            """
+                import difflib
+
+                import attr, os
+                import third_party
+            """,
+        )
+
     def test_merging_import_items(self) -> None:
         self.assertUsortResult(
             """

--- a/usort/types.py
+++ b/usort/types.py
@@ -218,7 +218,7 @@ class SortableImport:
 
         return results
 
-    def __attrs_post_init__(self) -> None:
+    def calculate_sort_key(self) -> None:
         top: Optional[str] = None
         ndots = 0
 
@@ -238,6 +238,9 @@ class SortableImport:
             is_from_import=bool(self.stem),
             ndots=ndots,
         )
+
+    def __attrs_post_init__(self) -> None:
+        self.calculate_sort_key()
 
 
 @dataclass(repr=False)

